h2spatial-ext/src/test/java/org/h2gis/h2spatialext/SpatialFunctionTest.java

```diff
@@ -129,6 +129,47 @@ public void test_ST_ExplodeEmptyGeometryCollection() throws Exception {
     }
     
     @Test
+    public void test_ST_ExplodeWithQuery() throws Exception {
+        st.execute("CREATE TABLE forests ( fid INTEGER NOT NULL PRIMARY KEY, name CHARACTER VARYING(64),"
+                + " boundary MULTIPOLYGON);"
+                + "INSERT INTO forests VALUES(109, 'Green Forest', ST_MPolyFromText( 'MULTIPOLYGON(((28 26,28 0,84 0,"
+                + "84 42,28 26), (52 18,66 23,73 9,48 6,52 18)),((59 18,67 18,67 13,59 13,59 18)))', 101));");
+        ResultSet rs = st.executeQuery("SELECT ST_AsText(boundary) FROM ST_Explode('select * from forests') WHERE name = 'Green Forest' and explod_id=2");
+        assertTrue(rs.next());
+        assertEquals("POLYGON ((59 18, 67 18, 67 13, 59 13, 59 18))", rs.getString(1));
+        st.execute("drop table forests");
+    }
+    
+    @Test(expected = NullPointerException.class)
```

Comment:
Your function should throw the original SQL exception in order to understand why the query fail.